test/spec/ol/tile.test.js

```diff
@@ -1,10 +1,72 @@
-
-
 goog.require('ol');
 goog.require('ol.ImageTile');
+goog.require('ol.Tile');
 goog.require('ol.TileState');
 
+
 describe('ol.Tile', function() {
+  describe('constructor', function()  {
+    it('sets a default transition', function() {
+      var coord = [0, 0, 0];
+      var tile = new ol.Tile(coord, ol.TileState.IDLE);
+      expect(tile.transition_).to.equal(275);
+    });
+
+    it('allows the transition to be set', function() {
+      var coord = [0, 0, 0];
+      var transition = 500;
+      var tile = new ol.Tile(coord, ol.TileState.IDLE, {transition: transition});
+      expect(tile.transition_).to.equal(transition);
+    });
+  });
+
+  describe('#getAlpha()', function() {
+    it('returns the alpha value for a tile in transition', function() {
+      var coord = [0, 0, 0];
+      var tile = new ol.Tile(coord, ol.TileState.IDLE);
+      var id = 'test';
+      var time = Date.now();
+
+      var startAlpha = tile.getAlpha(id, time);
+      expect(startAlpha > 0).to.be(true);
+      expect(startAlpha < 1).to.be(true);
+
+      time += tile.transition_ / 2;
+      var midAlpha = tile.getAlpha(id, time);
+      expect(midAlpha > startAlpha).to.be(true);
+      expect(midAlpha < 1).to.be(true);
+
+      time += tile.transition_ / 2;
+      var endAlpha = tile.getAlpha(id, time);
+      expect(endAlpha).to.be(1);
```

Comment:
Does this only pass because of the default of 275ms? It'd be nice if the default could particiate in the test, IMO. But I might also just be misreading it.

Reply:
This is true because we are calling`tile.getAlpha(id, time)` with a time delta that is equal to `tile.transition_`.  The assertion would pass if this value were something different than 275. We're not relying on any delta in clock time, just a delta in the `time` between the first call to `getAlpha` and this one.

Reply:
Got it.